File: pytyon/374. 猜数字大小.py
import sys
from collections import defaultdict
from collections import Counter
class Solution:
    def guessNumber(self, n: int) -> int:
        left = 1
        right = n

        while left < right:
            middle = (left + right) // 2

            if guess(middle) == -1:
                right = middle
            elif guess(middle) == 1:
                if left == middle:
                    return left
            
                left = middle

            elif:
                return middle


        return left



                
        
             
        
if __name__ == "__main__":
    solution = Solution()
    nums1 = "hello"
    m = "111"

    nums2 = [1,2,3]    
    n = 3

    result = solution.reverseVowels(nums1)

    #print(solution.ls)

    print(nums1, result)

File: c#/374. 猜数字大小.cs
using System;

namespace leetcode
{
    public class GuessGame {
        public int guess(int num) {
            var picker = 1702766719;

            if(num < picker) {
                return 1;
            }
            else if (num == picker) {
                return 0;
            }
            return -1;
        }
    }

    public class Solution374 : GuessGame {
       
        public int GuessNumber(int n) {
            var left = 1;
            var right = n;

            while(left < right) {
                var middle = left + (right - left)/ 2;

                Console.WriteLine($"{left}, {right}, {middle}");

                var temp = guess(middle); 
                if (temp == 0) {
                    return middle;
                } 
                else if (temp == -1) {
                    right = middle;
                }
                else {
                    if  (left == middle) {
                        left ++;
                    }
                    else {
                        left = middle;
                    }
                }
            }

            return left;
        }
    }
}